bftengine/include/bftengine/ControlStateManager.hpp

```diff
@@ -0,0 +1,42 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+#include <string>
+#include <unordered_map>
+
+namespace bftEngine {
+
+class IControlStateManager {
```

Comment:
I don't see a good reason to start off with an abstract interface here. It seems like premature abstraction to me. IMO there should only be a single, concrete, `ControlStateManager`, that is responsible for managing control state across reserved pages. Furthermore a single replica can only take a single instance of the `ControlStateManager`, so any functionality needed across components will have to use the same instance. The manager itself can allow plugins, but again this seems premature to me. We should start with the simplest thing and then get more complicated if we need it.

Reply:
The abstract class has been removed